src/types.rs

```diff
@@ -483,22 +483,16 @@ impl Rewrite for ast::Ty {
                 ty.rewrite(context, budget, offset + 1).map(|ty_str| format!("({})", ty_str))
             }
             ast::TyTup(ref tup_ret) => {
-                let inner = try_opt!(tup_ret.iter()
-                                            .map(|item| item.rewrite(context, width, offset))
-                                            .fold(Some("".to_owned()),
-                                                  |sum, x| {
-                                                      match (sum, x) {
-                                                          (Some(sum), Some(x)) => {
-                                                              if sum == "" {
-                                                                  // First item.
-                                                                  Some(x)
-                                                              } else {
-                                                                  Some(sum + ", " + &x)
-                                                              }
-                                                          }
-                                                          _ => None,
-                                                      }
-                                                  }));
+                let inner = if let [ref item] = &**tup_ret {
```

Comment:
Just a stylistic choice -- I'd have index into the vec after checking the length anyway, so I figured I'd do it all in one step. I don't feel strongly though.